File: algorithm/greedy/prb2/answer.c
import time

n, k = map(int, input().split())
start_time = time.time()
result = 0

while n>=k : 
  while n%k != 0:
    n -= 1
    result += 1
  n //= k
  result += 1

while n>1:
  n-=1
  result += 1

print(result)
end_time = time.time()

print("process time = ",end_time - start_time)

File: algorithm/binary search/pb3 떡볶이 떡 만들기/answer.py

import time

# n, m = map(int,input().split())
# length = list(map(int,input().split()))

n=4
m=6
length = [19,15,10,17]

start_time = time.time()

def binary_search(target,start,end):
  while start <= end:
    mid = (start + end) // 2
    total = 0
    
    for i in length:
      if i > mid:
        total += (i-mid)

    if total < target:
      end = mid - 1
    else:
      result = mid
      start = mid + 1
  return result

#log(N)*N^2

print(binary_search(m,0,max(length)))

end_time = time.time()
print(end_time - start_time)
